src/python/zquantum/core/circuit/conversions/pyquil_conversions.py

```diff
@@ -231,14 +235,54 @@ def convert_circuit_to_pyquil(
 
 
 @singledispatch
-def convert_from_pyquil(obj: Union[pyquil.Program, pyquil.quil.Gate]):
+def convert_from_pyquil(obj: Union[pyquil.Program, pyquil.quil.Gate], custom_gates=None):
     raise NotImplementedError(
         f"Conversion from pyquil to orquestra not implemented for {obj}"
     )
 
 
+def custom_gate_factory_from_pyquil_defgate(gate: pyquil.quil.DefGate):
+    num_qubits = int(math.log(gate.matrix.shape[0], 2))
```

Comment:
If this is all `math` is used for, you can use `np.log2` as well I believe. Definitely not a necessary change however

Reply:
Nice catch. I initially used numpy, but found out that it cannot handle arbitrarily large integers, which would put a limit on a circuit width. For instance `np.log2(2 ** 64)` already throws TypeError.